Backpack/CardButton/Classes/SaveCardButton.swift

```diff
@@ -0,0 +1,273 @@
+        guard viewConfigurator.shouldAddButtonBackground(style: style) else {
+            if wasAdded {
+                containedBackgroundCircle.removeFromSuperview()
+            }
+            return
+        }
+
+        guard !wasAdded else { return }
+        addSubview(containedBackgroundCircle)
+        let circleSize = viewConfigurator.buttonBackgroundSize(size: size)
+        containedBackgroundCircle.layer.cornerRadius = circleSize / 2
+
+        NSLayoutConstraint.activate([
+            containedBackgroundCircle.widthAnchor.constraint(equalToConstant: circleSize),
+            containedBackgroundCircle.heightAnchor.constraint(equalToConstant: circleSize),
+            containedBackgroundCircle.centerXAnchor.constraint(equalTo: centerXAnchor),
+            containedBackgroundCircle.centerYAnchor.constraint(equalTo: centerYAnchor)
+        ])
+        imageView.map { bringSubviewToFront($0) }
+    }
+
+    private func updateAccessibilityTraits() {
+        if checked {
+            accessibilityTraits.insert(.selected)
+        } else {
+            accessibilityTraits.remove(.selected)
+        }
+    }
+
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        guard traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle else { return }
+        updateLookAndFeel()
+    }
+}
+
+extension BPKSaveCardButton {
+    public func setChecked(_ newValue: Bool, animated: Bool) {
+        guard animated, let imageView = imageView else {
+            checked = newValue
+            return
+        }
+        if newValue {
+            // Update icon color before animation starts
+            checked = true
+            // 1. resize to 30px
+            // 2. wait
+            // 3. resize back to 24px
+            let transformBefore = imageView.transform
+            UIView.animate(
+                withDuration: BPKDuration.animationDurationSm,
+                delay: 0,
+                options: UIView.AnimationOptions([.beginFromCurrentState, .curveEaseOut])
+            ) {
+                imageView.transform = CGAffineTransformMakeScale(1.25, 1.25)
+            }
+            UIView.animate(
+                withDuration: BPKDuration.animationDurationSm,
+                delay: BPKDuration.animationDurationSm * 2,
+                options: UIView.AnimationOptions([.beginFromCurrentState, .curveEaseIn])
+            ) {
+                imageView.transform = transformBefore
+            }
+        } else {
+            // return to default state with transition
+            UIView.transition(
+                with: self,
+                duration: BPKDuration.animationDurationSm,
+                options: .transitionCrossDissolve
+            ) {
+                self.checked = newValue
+            }
+        }
+    }
+}
+
+private extension BPKSaveCardButton {
```

Comment:
Can this move to a separate BPKSaveCardButton+Style file?

Reply:
sure

Reply:
The new extension was created in 0d5636fc03c4bd4bfe15044efb24fa6fd6345153